pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
With the new approch I think we are missing the previous internally timeaxis=1.

Example
```
#Sender side
nb_channel = 16
nb_sample = 1000
sh_arr1 = SharedArray(shape=(nb_channel, nb_sample), dtype='float32', shm_id=None)
np_arr1 = sh_arr1.to_numpy() 
#we are wrong here because timaxis is not 0 so manually:
np_arr1 = np_arr1.swapaxes(0,1)
print(np_arr1.shape) #(1000, 16) OK
#receiver side
spec = pickle.dumps(orig_array.to_dict())
sh_arr2 = SharedArray(**pickle.loads(spec))
np_arr1 = sh_arr2.to_numpy() 
# do not known that we need swapaxes
print(np_arr2.shape) #(16, 1000)  BAD
```

I think we need something more magic No ?
This would lead to add 'swapaxes' key in streamspec.... (which is at the end the generalisation of timeaxis but for ND!!)

```
class SharedArray:
 def __init__(self, shape=(1,), dtype='float64', swapaxes=None, shm_id=None):
        self.shape = shape
        self.dtype = np.dtype(dtype)
        self.swapaxes = swapaxes
        nbytes = np.prod(shape)*self.dtype.itemsize
        self.shmem = SharedMem(nbytes, shm_id)
    
    def to_dict(self):
        return {'shape': self.shape, 'dtype': self.dtype, 'swapaxes':self.swapaxes,  'shm_id': self.shmem.shm_id}
    
    def to_numpy(self):
        np_arr = np.frombuffer(self.shmem.mmap, dtype=self.dtype).reshape(self.shape)
        if self.swapaxes is not None:
            np_arr = np_arr.swapaxes(self.np_arr)
        return np_arr


```